compiler/code-gen/declarations.cpp

```diff
@@ -810,7 +819,7 @@ static void compile_json_visitor_call(CodeGenerator &W, ClassPtr json_encoder, C
   }
 }
 
-void ClassDeclaration::compile_accept_json_visitor(CodeGenerator &W, ClassPtr klass, bool to_encode, ClassPtr json_encoder) {
+static void do_compile_accept_json_visitor(CodeGenerator &W, ClassPtr klass, bool to_encode, ClassPtr json_encoder, bool compile_declaration_only) {
```

Comment:
As I mentioned, there is even 2 `bool` parameters here. Maybe change `to_encode\to_decode` to `enum`, too?

Reply:
We use bool params almost everywhere for such cases